src/ReactionService.Business/Commands/Reaction/CreateReactionCommand.cs

```diff
@@ -0,0 +1,109 @@
+using LT.DigitalOffice.ReactionService.Data.Interfaces;
+using LT.DigitalOffice.ReactionService.Mappers.Db.Interfaces;
+using LT.DigitalOffice.ReactionService.Models.Dto.Requests;
+using LT.DigitalOffice.ReactionService.Validation.Reactions.Interfaces;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.ReactionService.Business.Commands.Reaction;
+
+public class CreateReactionCommand : ICreateReactionCommand
+{
+  private readonly ICreateReactionRequestValidator _validator;
+  private readonly IResponseCreator _responseCreator;
+  private readonly IAccessValidator _accessValidator;
+  private readonly IReactionRepository _reactionRepository;
+  private readonly IImageService _imageService;
+  private readonly IDbReactionMapper _mapper;
+  private readonly IHttpContextAccessor _httpContextAccessor;
+  private readonly IImageCompressHelper _imageCompressHelper;
+  private readonly IImageResizeHelper _imageResizeHelper;
+
+  public CreateReactionCommand(
+    ICreateReactionRequestValidator validator,
+    IResponseCreator responseCreator,
+    IAccessValidator accessValidator,
+    IReactionRepository reactionRepository,
+    IImageService imageService,
+    IDbReactionMapper mapper,
+    IHttpContextAccessor httpContextAccessor,
+    IImageCompressHelper imageCompressHelper,
+    IImageResizeHelper imageResizeHelper)
+  {
+    _validator = validator;
+    _responseCreator = responseCreator;
+    _accessValidator = accessValidator;
+    _reactionRepository = reactionRepository;
+    _imageService = imageService;
+    _mapper = mapper;
+    _httpContextAccessor = httpContextAccessor;
+    _imageCompressHelper = imageCompressHelper;
+    _imageResizeHelper = imageResizeHelper;
+  }
+
+  public async Task<OperationResultResponse<Guid?>> ExecuteAsync(CreateReactionRequest request)
+  {
+    if (!(await _accessValidator.IsAdminAsync(_httpContextAccessor.HttpContext.GetUserId())))
+    {
+      return _responseCreator.CreateFailureResponse<Guid?>(HttpStatusCode.Forbidden);
+    }
+
+    ValidationResult validationResult = await _validator.ValidateAsync(request);
+
+    if (!validationResult.IsValid)
+    {
+      return _responseCreator.CreateFailureResponse<Guid?>(
+        HttpStatusCode.BadRequest,
+        validationResult.Errors.Select(x => x.ErrorMessage).ToList());
+    }
+
+    OperationResultResponse<Guid?> response = new();
+
+    (bool isResizeSuccess, string imageContent, request.Extension) = await _imageResizeHelper.ResizeAsync(request.Content, request.Extension, 24);
+
+    if (!isResizeSuccess)
+    {
+      response.Errors.Add("Resize operation have been failed");
+    }
+
+    if ((imageContent is not null) && (Convert.FromBase64String(imageContent).Length / 1000 > 10))
+    {
+      (bool isCompressSuccess, imageContent, request.Extension) = await _imageCompressHelper.CompressAsync(request.Content, request.Extension, 10);
```

Comment:
if resize is success, resized image should be compressed

Reply:
Done